components/uniset/src/lib.rs

```diff
@@ -67,6 +68,7 @@ pub use utils::*;
 pub enum UnicodeSetError {
     InvalidSet(Vec<u32>),
     InvalidRange(u32, u32),
+    PropDataLoad,
```

Comment:
Rust has mechanisms for chaining errors together, so you should keep the old error when possible.

```suggestion
    PropDataLoad(DataError),
```

Reply:
I could use some advice on this. The reason I originally avoided adding that type argument to the enum variant is that I get this error:
```
error[E0423]: expected value, found enum `DataError`
  --> components/uniset/src/lib.rs:82:28
   |
82 |         Self::PropDataLoad(DataError)
   |                            ^^^^^^^^^
   |
help: you might have meant to use one of the following enum variants
   |
82 |         Self::PropDataLoad(DataError::NeedsTypeInfo)
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^
82 |         Self::PropDataLoad(DataError::MissingPayload)
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^
help: the following enum variants are available
   |
82 |         Self::PropDataLoad((DataError::UnsupportedCategory(/* fields */)))
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
82 |         Self::PropDataLoad((DataError::UnsupportedResourceKey(/* fields */)))
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
82 |         Self::PropDataLoad((DataError::UnavailableResourceOptions(/* fields */)))
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
82 |         Self::PropDataLoad((DataError::NeedsLanguageIdentifier(/* fields */)))
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     and 3 other candidates

error[E0369]: binary operation `==` cannot be applied to type `DataError`
  --> components/uniset/src/lib.rs:71:18
   |
71 |     PropDataLoad(DataError),
   |                  ^^^^^^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0369]: binary operation `!=` cannot be applied to type `DataError`
  --> components/uniset/src/lib.rs:71:18
   |
71 |     PropDataLoad(DataError),
   |                  ^^^^^^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
```



But when I add `PartialEq` to the `#derive` on DataError (aka the Error in the provider component), I then instead got these errors:
```
error[E0369]: binary operation `==` cannot be applied to type `erased_serde::Error`
  --> components/provider/src/error.rs:43:11
   |
43 |     Serde(erased_serde::Error),
   |           ^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0369]: binary operation `==` cannot be applied to type `Box<dyn StdError>`
  --> components/provider/src/error.rs:46:14
   |
46 |     Resource(Box<dyn std::error::Error>),
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0369]: binary operation `!=` cannot be applied to type `erased_serde::Error`
  --> components/provider/src/error.rs:43:11
   |
43 |     Serde(erased_serde::Error),
   |           ^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0369]: binary operation `!=` cannot be applied to type `Box<dyn StdError>`
  --> components/provider/src/error.rs:46:14
   |
46 |     Resource(Box<dyn std::error::Error>),
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
Done. Result of the offline convo: I needed to propagate the new `Cow<'s, str>` type through the code to other type signatures (including for the `name` field of `UnicodeProperty`